File: app/src/main/java/taxi/kassa/ui/adapters/AccountsAdapter.java
package taxi.kassa.ui.adapters;

import android.content.Context;
import android.content.DialogInterface;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ProgressBar;
import android.widget.TextView;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import taxi.kassa.R;
import taxi.kassa.responses.Account;
import taxi.kassa.responses.ResponseAPI;
import taxi.kassa.responses.ResponseSimple;

import static taxi.kassa.ui.MainActivity.api;

public class AccountsAdapter extends RecyclerView.Adapter<AccountsAdapter.ViewHolder> {

    private LayoutInflater inflater;
    private List<Account> accounts_list;
    private Context context;

    public AccountsAdapter(Context context, List<Account> accounts_list) {
        this.accounts_list = accounts_list;
        this.inflater = LayoutInflater.from(context);
        this.context = context;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        final TextView bank_name, account, driver_name;
        final ImageButton btn_detele;
        ViewHolder(View view){
            super(view);
            bank_name = view.findViewById(R.id.bank_name);
            account = view.findViewById(R.id.account);
            driver_name = view.findViewById(R.id.driver_name);
            btn_detele = view.findViewById(R.id.btn_delete);
        }
    }
    @NonNull
    @Override
    public AccountsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.account_item, parent, false);
        return new ViewHolder(view);
    }


    @Override
    public void onBindViewHolder(@NonNull AccountsAdapter.ViewHolder holder, final int position) {
        final Account info = accounts_list.get(position);
        final Boolean readonly = info.getReadonly();
        // info
        holder.bank_name.setText(info.getBankName());
        holder.account.setText("Р/с: " + info.getAccountNumber());
        holder.driver_name.setText(info.getDriverName());
        // btn delete
        if (!readonly) {
            holder.btn_detele.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    new AlertDialog.Builder(context)
                            //.setIcon(android.R.drawable.ic_dialog_alert)
                            .setTitle("Удаление счета")
                            .setMessage("Вы действительно хотите удалить этот счет?")
                            .setPositiveButton("Да", new DialogInterface.OnClickListener() {
                                @Override
                                public void onClick(DialogInterface dialog, int which) {
                                    int account_id = info.getId();
                                    // api
                                    Call<ResponseAPI<ResponseSimple>> call = api.deleteAccount(account_id);
                                    call.enqueue(new Callback<ResponseAPI<ResponseSimple>>() {
                                        @Override
                                        public void onResponse(@NonNull Call<ResponseAPI<ResponseSimple>> call, @NonNull Response<ResponseAPI<ResponseSimple>> response) {
                                            if (response.code() == 200 && response.body() != null) {
                                                accounts_list.remove(position);
                                                notifyItemRemoved(position);
                                                notifyItemRangeChanged(position, accounts_list.size());
                                            }
                                        }

                                        @Override
                                        public void onFailure(@NonNull Call<ResponseAPI<ResponseSimple>> call, @NonNull Throwable t) {
                                        }
                                    });
                                }

                            })
                            .setNegativeButton("Нет", null)
                            .show();
                }
            });
        } else {
            holder.btn_detele.setVisibility(ProgressBar.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return accounts_list.size();
    }

}

File: app/src/main/java/taxi/kassa/view/withdraws/withdraw_create/daily/AccountsAdapter.kt
package taxi.kassa.view.withdraws.withdraw_create.daily

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_account.view.*
import taxi.kassa.R
import taxi.kassa.model.responses.Account
import taxi.kassa.util.Constants.ALFABANK
import taxi.kassa.util.Constants.BINBANK
import taxi.kassa.util.Constants.SBERBANK
import taxi.kassa.util.Constants.TINKOFF
import taxi.kassa.util.Constants.VTB_BANK

class AccountsAdapter(
    private val accounts: List<Account>,
    private val clickListener: (Account, View) -> Unit
) : RecyclerView.Adapter<AccountsAdapter.AccountsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = AccountsViewHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_account, parent, false)
    )

    override fun getItemCount() = accounts.size

    override fun onBindViewHolder(holder: AccountsViewHolder, position: Int) {
        val account = accounts[position]

        with(holder) {
            bankIcon.setImageResource(
                when {
                    account.bankName.contains(ALFABANK, true) -> R.drawable.ic_alfa
                    account.bankName.contains(BINBANK, true) -> R.drawable.ic_binbank
                    account.bankName.contains(VTB_BANK, true) -> R.drawable.ic_vtb
                    account.bankName.contains(SBERBANK, true) -> R.drawable.ic_sberbank
                    account.bankName.contains(TINKOFF, true) -> R.drawable.ic_tinkoff
                    else -> R.drawable.transparent
                }
            )
            bankName.text = account.bankName
            accountNumber.text = itemView.context.getString(R.string.order_format, account.accountNumber)
            name.text = account.driverName

            itemView.setOnClickListener { clickListener(account, itemView) }
        }
    }

    class AccountsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val background: ImageView = view.account_background
        val bankIcon: ImageView = view.bank_icon
        val bankName: TextView = view.bank_name_tv
        val accountNumber: TextView = view.account_number
        val name: TextView = view.name_tv
    }
}